// react-native-agora/lib/commonjs/impl/IAgoraMediaPlayerImpl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IMediaPlayerImpl = exports.IMediaPlayerCacheManagerImpl = void 0;
exports.processIMediaPlayerVideoFrameObserver = processIMediaPlayerVideoFrameObserver;
var _AgoraMediaBase = require("../AgoraMediaBase");
var _IrisApiEngine = require("../internal/IrisApiEngine");
// @ts-ignore
class IMediaPlayerImpl {
  getMediaPlayerId() {
    const apiType = this.getApiTypeFromGetMediaPlayerId();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMediaPlayerId() {
    return 'MediaPlayer_getMediaPlayerId';
  }
  open(url, startPos) {
    const apiType = this.getApiTypeFromOpen(url, startPos);
    const jsonParams = {
      url: url,
      startPos: startPos,
      toJSON: () => {
        return {
          url: url,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromOpen(url, startPos) {
    return 'MediaPlayer_open';
  }
  openWithMediaSource(source) {
    const apiType = this.getApiTypeFromOpenWithMediaSource(source);
    const jsonParams = {
      source: source,
      toJSON: () => {
        return {
          source: source
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromOpenWithMediaSource(source) {
    return 'MediaPlayer_openWithMediaSource';
  }
  play() {
    const apiType = this.getApiTypeFromPlay();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPlay() {
    return 'MediaPlayer_play';
  }
  pause() {
    const apiType = this.getApiTypeFromPause();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPause() {
    return 'MediaPlayer_pause';
  }
  stop() {
    const apiType = this.getApiTypeFromStop();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStop() {
    return 'MediaPlayer_stop';
  }
  resume() {
    const apiType = this.getApiTypeFromResume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResume() {
    return 'MediaPlayer_resume';
  }
  seek(newPos) {
    const apiType = this.getApiTypeFromSeek(newPos);
    const jsonParams = {
      newPos: newPos,
      toJSON: () => {
        return {
          newPos: newPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSeek(newPos) {
    return 'MediaPlayer_seek';
  }
  setAudioPitch(pitch) {
    const apiType = this.getApiTypeFromSetAudioPitch(pitch);
    const jsonParams = {
      pitch: pitch,
      toJSON: () => {
        return {
          pitch: pitch
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioPitch(pitch) {
    return 'MediaPlayer_setAudioPitch';
  }
  getDuration() {
    const apiType = this.getApiTypeFromGetDuration();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const duration = jsonResults.duration;
    return duration;
  }
  getApiTypeFromGetDuration() {
    return 'MediaPlayer_getDuration';
  }
  getPlayPosition() {
    const apiType = this.getApiTypeFromGetPlayPosition();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const pos = jsonResults.pos;
    return pos;
  }
  getApiTypeFromGetPlayPosition() {
    return 'MediaPlayer_getPlayPosition';
  }
  getStreamCount() {
    const apiType = this.getApiTypeFromGetStreamCount();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const count = jsonResults.count;
    return count;
  }
  getApiTypeFromGetStreamCount() {
    return 'MediaPlayer_getStreamCount';
  }
  getStreamInfo(index) {
    const apiType = this.getApiTypeFromGetStreamInfo(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const info = jsonResults.info;
    return info;
  }
  getApiTypeFromGetStreamInfo(index) {
    return 'MediaPlayer_getStreamInfo';
  }
  setLoopCount(loopCount) {
    const apiType = this.getApiTypeFromSetLoopCount(loopCount);
    const jsonParams = {
      loopCount: loopCount,
      toJSON: () => {
        return {
          loopCount: loopCount
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetLoopCount(loopCount) {
    return 'MediaPlayer_setLoopCount';
  }
  setPlaybackSpeed(speed) {
    const apiType = this.getApiTypeFromSetPlaybackSpeed(speed);
    const jsonParams = {
      speed: speed,
      toJSON: () => {
        return {
          speed: speed
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlaybackSpeed(speed) {
    return 'MediaPlayer_setPlaybackSpeed';
  }
  selectAudioTrack(index) {
    const apiType = this.getApiTypeFromSelectAudioTrack(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSelectAudioTrack(index) {
    return 'MediaPlayer_selectAudioTrack';
  }
  setPlayerOptionInInt(key, value) {
    const apiType = this.getApiTypeFromSetPlayerOptionInInt(key, value);
    const jsonParams = {
      key: key,
      value: value,
      toJSON: () => {
        return {
          key: key,
          value: value
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlayerOptionInInt(key, value) {
    return 'MediaPlayer_setPlayerOptionInInt';
  }
  setPlayerOptionInString(key, value) {
    const apiType = this.getApiTypeFromSetPlayerOptionInString(key, value);
    const jsonParams = {
      key: key,
      value: value,
      toJSON: () => {
        return {
          key: key,
          value: value
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetPlayerOptionInString(key, value) {
    return 'MediaPlayer_setPlayerOptionInString';
  }
  takeScreenshot(filename) {
    const apiType = this.getApiTypeFromTakeScreenshot(filename);
    const jsonParams = {
      filename: filename,
      toJSON: () => {
        return {
          filename: filename
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromTakeScreenshot(filename) {
    return 'MediaPlayer_takeScreenshot';
  }
  selectInternalSubtitle(index) {
    const apiType = this.getApiTypeFromSelectInternalSubtitle(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSelectInternalSubtitle(index) {
    return 'MediaPlayer_selectInternalSubtitle';
  }
  setExternalSubtitle(url) {
    const apiType = this.getApiTypeFromSetExternalSubtitle(url);
    const jsonParams = {
      url: url,
      toJSON: () => {
        return {
          url: url
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetExternalSubtitle(url) {
    return 'MediaPlayer_setExternalSubtitle';
  }
  getState() {
    const apiType = this.getApiTypeFromGetState();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetState() {
    return 'MediaPlayer_getState';
  }
  mute(muted) {
    const apiType = this.getApiTypeFromMute(muted);
    const jsonParams = {
      muted: muted,
      toJSON: () => {
        return {
          muted: muted
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMute(muted) {
    return 'MediaPlayer_mute';
  }
  getMute() {
    const apiType = this.getApiTypeFromGetMute();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const muted = jsonResults.muted;
    return muted;
  }
  getApiTypeFromGetMute() {
    return 'MediaPlayer_getMute';
  }
  adjustPlayoutVolume(volume) {
    const apiType = this.getApiTypeFromAdjustPlayoutVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustPlayoutVolume(volume) {
    return 'MediaPlayer_adjustPlayoutVolume';
  }
  getPlayoutVolume() {
    const apiType = this.getApiTypeFromGetPlayoutVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const volume = jsonResults.volume;
    return volume;
  }
  getApiTypeFromGetPlayoutVolume() {
    return 'MediaPlayer_getPlayoutVolume';
  }
  adjustPublishSignalVolume(volume) {
    const apiType = this.getApiTypeFromAdjustPublishSignalVolume(volume);
    const jsonParams = {
      volume: volume,
      toJSON: () => {
        return {
          volume: volume
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustPublishSignalVolume(volume) {
    return 'MediaPlayer_adjustPublishSignalVolume';
  }
  getPublishSignalVolume() {
    const apiType = this.getApiTypeFromGetPublishSignalVolume();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const volume = jsonResults.volume;
    return volume;
  }
  getApiTypeFromGetPublishSignalVolume() {
    return 'MediaPlayer_getPublishSignalVolume';
  }
  setView(view) {
    const apiType = this.getApiTypeFromSetView(view);
    const jsonParams = {
      view: view,
      toJSON: () => {
        return {
          view: view
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetView(view) {
    return 'MediaPlayer_setView';
  }
  setRenderMode(renderMode) {
    const apiType = this.getApiTypeFromSetRenderMode(renderMode);
    const jsonParams = {
      renderMode: renderMode,
      toJSON: () => {
        return {
          renderMode: renderMode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRenderMode(renderMode) {
    return 'MediaPlayer_setRenderMode';
  }
  registerPlayerSourceObserver(observer) {
    const apiType = this.getApiTypeFromRegisterPlayerSourceObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterPlayerSourceObserver(observer) {
    return 'MediaPlayer_registerPlayerSourceObserver';
  }
  unregisterPlayerSourceObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterPlayerSourceObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterPlayerSourceObserver(observer) {
    return 'MediaPlayer_unregisterPlayerSourceObserver';
  }
  registerAudioFrameObserver(observer) {
    let mode = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.RawAudioFrameOpModeType.RawAudioFrameOpModeReadOnly;
    const apiType = this.getApiTypeFromRegisterAudioFrameObserver(observer, mode);
    const jsonParams = {
      observer: observer,
      mode: mode,
      toJSON: () => {
        return {
          mode: mode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterAudioFrameObserver(observer) {
    let mode = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _AgoraMediaBase.RawAudioFrameOpModeType.RawAudioFrameOpModeReadOnly;
    return 'MediaPlayer_registerAudioFrameObserver';
  }
  unregisterAudioFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterAudioFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterAudioFrameObserver(observer) {
    return 'MediaPlayer_unregisterAudioFrameObserver';
  }
  registerVideoFrameObserver(observer) {
    const apiType = this.getApiTypeFromRegisterVideoFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterVideoFrameObserver(observer) {
    return 'MediaPlayer_registerVideoFrameObserver';
  }
  unregisterVideoFrameObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterVideoFrameObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterVideoFrameObserver(observer) {
    return 'MediaPlayer_unregisterVideoFrameObserver';
  }
  registerMediaPlayerAudioSpectrumObserver(observer, intervalInMS) {
    const apiType = this.getApiTypeFromRegisterMediaPlayerAudioSpectrumObserver(observer, intervalInMS);
    const jsonParams = {
      observer: observer,
      intervalInMS: intervalInMS,
      toJSON: () => {
        return {
          intervalInMS: intervalInMS
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRegisterMediaPlayerAudioSpectrumObserver(observer, intervalInMS) {
    return 'MediaPlayer_registerMediaPlayerAudioSpectrumObserver';
  }
  unregisterMediaPlayerAudioSpectrumObserver(observer) {
    const apiType = this.getApiTypeFromUnregisterMediaPlayerAudioSpectrumObserver(observer);
    const jsonParams = {
      observer: observer,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnregisterMediaPlayerAudioSpectrumObserver(observer) {
    return 'MediaPlayer_unregisterMediaPlayerAudioSpectrumObserver';
  }
  setAudioDualMonoMode(mode) {
    const apiType = this.getApiTypeFromSetAudioDualMonoMode(mode);
    const jsonParams = {
      mode: mode,
      toJSON: () => {
        return {
          mode: mode
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetAudioDualMonoMode(mode) {
    return 'MediaPlayer_setAudioDualMonoMode';
  }
  getPlayerSdkVersion() {
    const apiType = this.getApiTypeFromGetPlayerSdkVersion();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetPlayerSdkVersion() {
    return 'MediaPlayer_getPlayerSdkVersion';
  }
  getPlaySrc() {
    const apiType = this.getApiTypeFromGetPlaySrc();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetPlaySrc() {
    return 'MediaPlayer_getPlaySrc';
  }
  openWithAgoraCDNSrc(src, startPos) {
    const apiType = this.getApiTypeFromOpenWithAgoraCDNSrc(src, startPos);
    const jsonParams = {
      src: src,
      startPos: startPos,
      toJSON: () => {
        return {
          src: src,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromOpenWithAgoraCDNSrc(src, startPos) {
    return 'MediaPlayer_openWithAgoraCDNSrc';
  }
  getAgoraCDNLineCount() {
    const apiType = this.getApiTypeFromGetAgoraCDNLineCount();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetAgoraCDNLineCount() {
    return 'MediaPlayer_getAgoraCDNLineCount';
  }
  switchAgoraCDNLineByIndex(index) {
    const apiType = this.getApiTypeFromSwitchAgoraCDNLineByIndex(index);
    const jsonParams = {
      index: index,
      toJSON: () => {
        return {
          index: index
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSwitchAgoraCDNLineByIndex(index) {
    return 'MediaPlayer_switchAgoraCDNLineByIndex';
  }
  getCurrentAgoraCDNIndex() {
    const apiType = this.getApiTypeFromGetCurrentAgoraCDNIndex();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCurrentAgoraCDNIndex() {
    return 'MediaPlayer_getCurrentAgoraCDNIndex';
  }
  enableAutoSwitchAgoraCDN(enable) {
    const apiType = this.getApiTypeFromEnableAutoSwitchAgoraCDN(enable);
    const jsonParams = {
      enable: enable,
      toJSON: () => {
        return {
          enable: enable
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAutoSwitchAgoraCDN(enable) {
    return 'MediaPlayer_enableAutoSwitchAgoraCDN';
  }
  renewAgoraCDNSrcToken(token, ts) {
    const apiType = this.getApiTypeFromRenewAgoraCDNSrcToken(token, ts);
    const jsonParams = {
      token: token,
      ts: ts,
      toJSON: () => {
        return {
          token: token,
          ts: ts
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRenewAgoraCDNSrcToken(token, ts) {
    return 'MediaPlayer_renewAgoraCDNSrcToken';
  }
  switchAgoraCDNSrc(src) {
    let syncPts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    const apiType = this.getApiTypeFromSwitchAgoraCDNSrc(src, syncPts);
    const jsonParams = {
      src: src,
      syncPts: syncPts,
      toJSON: () => {
        return {
          src: src,
          syncPts: syncPts
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSwitchAgoraCDNSrc(src) {
    let syncPts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;
    return 'MediaPlayer_switchAgoraCDNSrc';
  }
  switchSrc(src) {
    let syncPts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : true;
    const apiType = this.getApiTypeFromSwitchSrc(src, syncPts);
    const jsonParams = {
      src: src,
      syncPts: syncPts,
      toJSON: () => {
        return {
          src: src,
          syncPts: syncPts
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSwitchSrc(src) {
    let syncPts = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : true;
    return 'MediaPlayer_switchSrc';
  }
  preloadSrc(src, startPos) {
    const apiType = this.getApiTypeFromPreloadSrc(src, startPos);
    const jsonParams = {
      src: src,
      startPos: startPos,
      toJSON: () => {
        return {
          src: src,
          startPos: startPos
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPreloadSrc(src, startPos) {
    return 'MediaPlayer_preloadSrc';
  }
  playPreloadedSrc(src) {
    const apiType = this.getApiTypeFromPlayPreloadedSrc(src);
    const jsonParams = {
      src: src,
      toJSON: () => {
        return {
          src: src
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPlayPreloadedSrc(src) {
    return 'MediaPlayer_playPreloadedSrc';
  }
  unloadSrc(src) {
    const apiType = this.getApiTypeFromUnloadSrc(src);
    const jsonParams = {
      src: src,
      toJSON: () => {
        return {
          src: src
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUnloadSrc(src) {
    return 'MediaPlayer_unloadSrc';
  }
  setSpatialAudioParams(params) {
    const apiType = this.getApiTypeFromSetSpatialAudioParams(params);
    const jsonParams = {
      params: params,
      toJSON: () => {
        return {
          params: params
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSpatialAudioParams(params) {
    return 'MediaPlayer_setSpatialAudioParams';
  }
  setSoundPositionParams(pan, gain) {
    const apiType = this.getApiTypeFromSetSoundPositionParams(pan, gain);
    const jsonParams = {
      pan: pan,
      gain: gain,
      toJSON: () => {
        return {
          pan: pan,
          gain: gain
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSoundPositionParams(pan, gain) {
    return 'MediaPlayer_setSoundPositionParams';
  }
}

// @ts-ignore
exports.IMediaPlayerImpl = IMediaPlayerImpl;
class IMediaPlayerCacheManagerImpl {
  removeAllCaches() {
    const apiType = this.getApiTypeFromRemoveAllCaches();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRemoveAllCaches() {
    return 'MediaPlayerCacheManager_removeAllCaches';
  }
  removeOldCache() {
    const apiType = this.getApiTypeFromRemoveOldCache();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRemoveOldCache() {
    return 'MediaPlayerCacheManager_removeOldCache';
  }
  removeCacheByUri(uri) {
    const apiType = this.getApiTypeFromRemoveCacheByUri(uri);
    const jsonParams = {
      uri: uri,
      toJSON: () => {
        return {
          uri: uri
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromRemoveCacheByUri(uri) {
    return 'MediaPlayerCacheManager_removeCacheByUri';
  }
  setCacheDir(path) {
    const apiType = this.getApiTypeFromSetCacheDir(path);
    const jsonParams = {
      path: path,
      toJSON: () => {
        return {
          path: path
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetCacheDir(path) {
    return 'MediaPlayerCacheManager_setCacheDir';
  }
  setMaxCacheFileCount(count) {
    const apiType = this.getApiTypeFromSetMaxCacheFileCount(count);
    const jsonParams = {
      count: count,
      toJSON: () => {
        return {
          count: count
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetMaxCacheFileCount(count) {
    return 'MediaPlayerCacheManager_setMaxCacheFileCount';
  }
  setMaxCacheFileSize(cacheSize) {
    const apiType = this.getApiTypeFromSetMaxCacheFileSize(cacheSize);
    const jsonParams = {
      cacheSize: cacheSize,
      toJSON: () => {
        return {
          cacheSize: cacheSize
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetMaxCacheFileSize(cacheSize) {
    return 'MediaPlayerCacheManager_setMaxCacheFileSize';
  }
  enableAutoRemoveCache(enable) {
    const apiType = this.getApiTypeFromEnableAutoRemoveCache(enable);
    const jsonParams = {
      enable: enable,
      toJSON: () => {
        return {
          enable: enable
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAutoRemoveCache(enable) {
    return 'MediaPlayerCacheManager_enableAutoRemoveCache';
  }
  getCacheDir(length) {
    const apiType = this.getApiTypeFromGetCacheDir(length);
    const jsonParams = {
      length: length,
      toJSON: () => {
        return {
          length: length
        };
      }
    };
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    const path = jsonResults.path;
    return path;
  }
  getApiTypeFromGetCacheDir(length) {
    return 'MediaPlayerCacheManager_getCacheDir';
  }
  getMaxCacheFileCount() {
    const apiType = this.getApiTypeFromGetMaxCacheFileCount();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMaxCacheFileCount() {
    return 'MediaPlayerCacheManager_getMaxCacheFileCount';
  }
  getMaxCacheFileSize() {
    const apiType = this.getApiTypeFromGetMaxCacheFileSize();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetMaxCacheFileSize() {
    return 'MediaPlayerCacheManager_getMaxCacheFileSize';
  }
  getCacheFileCount() {
    const apiType = this.getApiTypeFromGetCacheFileCount();
    const jsonParams = {};
    const jsonResults = _IrisApiEngine.callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetCacheFileCount() {
    return 'MediaPlayerCacheManager_getCacheFileCount';
  }
}
exports.IMediaPlayerCacheManagerImpl = IMediaPlayerCacheManagerImpl;
function processIMediaPlayerVideoFrameObserver(handler, event, jsonParams) {
  switch (event) {
    case 'onFrame':
      if (handler.onFrame !== undefined) {
        handler.onFrame(jsonParams.frame);
      }
      break;
  }
}
