// react-native-agora/lib/module/impl/IAgoraMediaRecorderImpl.js
// @ts-ignore
export class IMediaRecorderImpl {
  setMediaRecorderObserver(callback) {
    const apiType = this.getApiTypeFromSetMediaRecorderObserver(callback);
    const jsonParams = {
      callback: callback,
      toJSON: () => {
        return {};
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetMediaRecorderObserver(callback) {
    return 'MediaRecorder_setMediaRecorderObserver';
  }
  startRecording(config) {
    const apiType = this.getApiTypeFromStartRecording(config);
    const jsonParams = {
      config: config,
      toJSON: () => {
        return {
          config: config
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRecording(config) {
    return 'MediaRecorder_startRecording';
  }
  stopRecording() {
    const apiType = this.getApiTypeFromStopRecording();
    const jsonParams = {};
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopRecording() {
    return 'MediaRecorder_stopRecording';
  }
}
import { callIrisApi } from '../internal/IrisApiEngine';
