// react-native-agora/lib/module/impl/IAgoraRtcEngineExImpl.js
import { IRtcEngineImpl } from './IAgoraRtcEngineImpl';

// @ts-ignore
export class IRtcEngineExImpl extends IRtcEngineImpl {
  joinChannelEx(token, connection, options) {
    const apiType = this.getApiTypeFromJoinChannelEx(token, connection, options);
    const jsonParams = {
      token: token,
      connection: connection,
      options: options,
      toJSON: () => {
        return {
          token: token,
          connection: connection,
          options: options
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromJoinChannelEx(token, connection, options) {
    return 'RtcEngineEx_joinChannelEx';
  }
  leaveChannelEx(connection, options) {
    const apiType = this.getApiTypeFromLeaveChannelEx(connection, options);
    const jsonParams = {
      connection: connection,
      options: options,
      toJSON: () => {
        return {
          connection: connection,
          options: options
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromLeaveChannelEx(connection, options) {
    return 'RtcEngineEx_leaveChannelEx';
  }
  updateChannelMediaOptionsEx(options, connection) {
    const apiType = this.getApiTypeFromUpdateChannelMediaOptionsEx(options, connection);
    const jsonParams = {
      options: options,
      connection: connection,
      toJSON: () => {
        return {
          options: options,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateChannelMediaOptionsEx(options, connection) {
    return 'RtcEngineEx_updateChannelMediaOptionsEx';
  }
  setVideoEncoderConfigurationEx(config, connection) {
    const apiType = this.getApiTypeFromSetVideoEncoderConfigurationEx(config, connection);
    const jsonParams = {
      config: config,
      connection: connection,
      toJSON: () => {
        return {
          config: config,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetVideoEncoderConfigurationEx(config, connection) {
    return 'RtcEngineEx_setVideoEncoderConfigurationEx';
  }
  setupRemoteVideoEx(canvas, connection) {
    const apiType = this.getApiTypeFromSetupRemoteVideoEx(canvas, connection);
    const jsonParams = {
      canvas: canvas,
      connection: connection,
      toJSON: () => {
        return {
          canvas: canvas,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetupRemoteVideoEx(canvas, connection) {
    return 'RtcEngineEx_setupRemoteVideoEx';
  }
  muteRemoteAudioStreamEx(uid, mute, connection) {
    const apiType = this.getApiTypeFromMuteRemoteAudioStreamEx(uid, mute, connection);
    const jsonParams = {
      uid: uid,
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRemoteAudioStreamEx(uid, mute, connection) {
    return 'RtcEngineEx_muteRemoteAudioStreamEx';
  }
  muteRemoteVideoStreamEx(uid, mute, connection) {
    const apiType = this.getApiTypeFromMuteRemoteVideoStreamEx(uid, mute, connection);
    const jsonParams = {
      uid: uid,
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRemoteVideoStreamEx(uid, mute, connection) {
    return 'RtcEngineEx_muteRemoteVideoStreamEx';
  }
  setRemoteVideoStreamTypeEx(uid, streamType, connection) {
    const apiType = this.getApiTypeFromSetRemoteVideoStreamTypeEx(uid, streamType, connection);
    const jsonParams = {
      uid: uid,
      streamType: streamType,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          streamType: streamType,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVideoStreamTypeEx(uid, streamType, connection) {
    return 'RtcEngineEx_setRemoteVideoStreamTypeEx';
  }
  muteLocalAudioStreamEx(mute, connection) {
    const apiType = this.getApiTypeFromMuteLocalAudioStreamEx(mute, connection);
    const jsonParams = {
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteLocalAudioStreamEx(mute, connection) {
    return 'RtcEngineEx_muteLocalAudioStreamEx';
  }
  muteLocalVideoStreamEx(mute, connection) {
    const apiType = this.getApiTypeFromMuteLocalVideoStreamEx(mute, connection);
    const jsonParams = {
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteLocalVideoStreamEx(mute, connection) {
    return 'RtcEngineEx_muteLocalVideoStreamEx';
  }
  muteAllRemoteAudioStreamsEx(mute, connection) {
    const apiType = this.getApiTypeFromMuteAllRemoteAudioStreamsEx(mute, connection);
    const jsonParams = {
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteAllRemoteAudioStreamsEx(mute, connection) {
    return 'RtcEngineEx_muteAllRemoteAudioStreamsEx';
  }
  muteAllRemoteVideoStreamsEx(mute, connection) {
    const apiType = this.getApiTypeFromMuteAllRemoteVideoStreamsEx(mute, connection);
    const jsonParams = {
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteAllRemoteVideoStreamsEx(mute, connection) {
    return 'RtcEngineEx_muteAllRemoteVideoStreamsEx';
  }
  setSubscribeAudioBlocklistEx(uidList, uidNumber, connection) {
    const apiType = this.getApiTypeFromSetSubscribeAudioBlocklistEx(uidList, uidNumber, connection);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      connection: connection,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeAudioBlocklistEx(uidList, uidNumber, connection) {
    return 'RtcEngineEx_setSubscribeAudioBlocklistEx';
  }
  setSubscribeAudioAllowlistEx(uidList, uidNumber, connection) {
    const apiType = this.getApiTypeFromSetSubscribeAudioAllowlistEx(uidList, uidNumber, connection);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      connection: connection,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeAudioAllowlistEx(uidList, uidNumber, connection) {
    return 'RtcEngineEx_setSubscribeAudioAllowlistEx';
  }
  setSubscribeVideoBlocklistEx(uidList, uidNumber, connection) {
    const apiType = this.getApiTypeFromSetSubscribeVideoBlocklistEx(uidList, uidNumber, connection);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      connection: connection,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeVideoBlocklistEx(uidList, uidNumber, connection) {
    return 'RtcEngineEx_setSubscribeVideoBlocklistEx';
  }
  setSubscribeVideoAllowlistEx(uidList, uidNumber, connection) {
    const apiType = this.getApiTypeFromSetSubscribeVideoAllowlistEx(uidList, uidNumber, connection);
    const jsonParams = {
      uidList: uidList,
      uidNumber: uidNumber,
      connection: connection,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNumber: uidNumber,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetSubscribeVideoAllowlistEx(uidList, uidNumber, connection) {
    return 'RtcEngineEx_setSubscribeVideoAllowlistEx';
  }
  setRemoteVideoSubscriptionOptionsEx(uid, options, connection) {
    const apiType = this.getApiTypeFromSetRemoteVideoSubscriptionOptionsEx(uid, options, connection);
    const jsonParams = {
      uid: uid,
      options: options,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          options: options,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVideoSubscriptionOptionsEx(uid, options, connection) {
    return 'RtcEngineEx_setRemoteVideoSubscriptionOptionsEx';
  }
  setRemoteVoicePositionEx(uid, pan, gain, connection) {
    const apiType = this.getApiTypeFromSetRemoteVoicePositionEx(uid, pan, gain, connection);
    const jsonParams = {
      uid: uid,
      pan: pan,
      gain: gain,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          pan: pan,
          gain: gain,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteVoicePositionEx(uid, pan, gain, connection) {
    return 'RtcEngineEx_setRemoteVoicePositionEx';
  }
  setRemoteUserSpatialAudioParamsEx(uid, params, connection) {
    const apiType = this.getApiTypeFromSetRemoteUserSpatialAudioParamsEx(uid, params, connection);
    const jsonParams = {
      uid: uid,
      params: params,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          params: params,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteUserSpatialAudioParamsEx(uid, params, connection) {
    return 'RtcEngineEx_setRemoteUserSpatialAudioParamsEx';
  }
  setRemoteRenderModeEx(uid, renderMode, mirrorMode, connection) {
    const apiType = this.getApiTypeFromSetRemoteRenderModeEx(uid, renderMode, mirrorMode, connection);
    const jsonParams = {
      uid: uid,
      renderMode: renderMode,
      mirrorMode: mirrorMode,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          renderMode: renderMode,
          mirrorMode: mirrorMode,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetRemoteRenderModeEx(uid, renderMode, mirrorMode, connection) {
    return 'RtcEngineEx_setRemoteRenderModeEx';
  }
  enableLoopbackRecordingEx(connection, enabled, deviceName) {
    const apiType = this.getApiTypeFromEnableLoopbackRecordingEx(connection, enabled, deviceName);
    const jsonParams = {
      connection: connection,
      enabled: enabled,
      deviceName: deviceName,
      toJSON: () => {
        return {
          connection: connection,
          enabled: enabled,
          deviceName: deviceName
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableLoopbackRecordingEx(connection, enabled, deviceName) {
    return 'RtcEngineEx_enableLoopbackRecordingEx';
  }
  adjustRecordingSignalVolumeEx(volume, connection) {
    const apiType = this.getApiTypeFromAdjustRecordingSignalVolumeEx(volume, connection);
    const jsonParams = {
      volume: volume,
      connection: connection,
      toJSON: () => {
        return {
          volume: volume,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustRecordingSignalVolumeEx(volume, connection) {
    return 'RtcEngineEx_adjustRecordingSignalVolumeEx';
  }
  muteRecordingSignalEx(mute, connection) {
    const apiType = this.getApiTypeFromMuteRecordingSignalEx(mute, connection);
    const jsonParams = {
      mute: mute,
      connection: connection,
      toJSON: () => {
        return {
          mute: mute,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromMuteRecordingSignalEx(mute, connection) {
    return 'RtcEngineEx_muteRecordingSignalEx';
  }
  adjustUserPlaybackSignalVolumeEx(uid, volume, connection) {
    const apiType = this.getApiTypeFromAdjustUserPlaybackSignalVolumeEx(uid, volume, connection);
    const jsonParams = {
      uid: uid,
      volume: volume,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          volume: volume,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAdjustUserPlaybackSignalVolumeEx(uid, volume, connection) {
    return 'RtcEngineEx_adjustUserPlaybackSignalVolumeEx';
  }
  getConnectionStateEx(connection) {
    const apiType = this.getApiTypeFromGetConnectionStateEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromGetConnectionStateEx(connection) {
    return 'RtcEngineEx_getConnectionStateEx';
  }
  enableEncryptionEx(connection, enabled, config) {
    const apiType = this.getApiTypeFromEnableEncryptionEx(connection, enabled, config);
    const jsonParams = {
      connection: connection,
      enabled: enabled,
      config: config,
      toJSON: () => {
        return {
          connection: connection,
          enabled: enabled,
          config: config
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableEncryptionEx(connection, enabled, config) {
    return 'RtcEngineEx_enableEncryptionEx';
  }
  createDataStreamEx(config, connection) {
    const apiType = this.getApiTypeFromCreateDataStreamEx(config, connection);
    const jsonParams = {
      config: config,
      connection: connection,
      toJSON: () => {
        return {
          config: config,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const streamId = jsonResults.streamId;
    return streamId;
  }
  getApiTypeFromCreateDataStreamEx(config, connection) {
    return 'RtcEngineEx_createDataStreamEx';
  }
  sendStreamMessageEx(streamId, data, length, connection) {
    const apiType = this.getApiTypeFromSendStreamMessageEx(streamId, data, length, connection);
    const jsonParams = {
      streamId: streamId,
      data: data,
      length: length,
      connection: connection,
      toJSON: () => {
        return {
          streamId: streamId,
          length: length,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendStreamMessageEx(streamId, data, length, connection) {
    return 'RtcEngineEx_sendStreamMessageEx';
  }
  addVideoWatermarkEx(watermarkUrl, options, connection) {
    const apiType = this.getApiTypeFromAddVideoWatermarkEx(watermarkUrl, options, connection);
    const jsonParams = {
      watermarkUrl: watermarkUrl,
      options: options,
      connection: connection,
      toJSON: () => {
        return {
          watermarkUrl: watermarkUrl,
          options: options,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromAddVideoWatermarkEx(watermarkUrl, options, connection) {
    return 'RtcEngineEx_addVideoWatermarkEx';
  }
  clearVideoWatermarkEx(connection) {
    const apiType = this.getApiTypeFromClearVideoWatermarkEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromClearVideoWatermarkEx(connection) {
    return 'RtcEngineEx_clearVideoWatermarkEx';
  }
  sendCustomReportMessageEx(id, category, event, label, value, connection) {
    const apiType = this.getApiTypeFromSendCustomReportMessageEx(id, category, event, label, value, connection);
    const jsonParams = {
      id: id,
      category: category,
      event: event,
      label: label,
      value: value,
      connection: connection,
      toJSON: () => {
        return {
          id: id,
          category: category,
          event: event,
          label: label,
          value: value,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSendCustomReportMessageEx(id, category, event, label, value, connection) {
    return 'RtcEngineEx_sendCustomReportMessageEx';
  }
  enableAudioVolumeIndicationEx(interval, smooth, reportVad, connection) {
    const apiType = this.getApiTypeFromEnableAudioVolumeIndicationEx(interval, smooth, reportVad, connection);
    const jsonParams = {
      interval: interval,
      smooth: smooth,
      reportVad: reportVad,
      connection: connection,
      toJSON: () => {
        return {
          interval: interval,
          smooth: smooth,
          reportVad: reportVad,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableAudioVolumeIndicationEx(interval, smooth, reportVad, connection) {
    return 'RtcEngineEx_enableAudioVolumeIndicationEx';
  }
  startRtmpStreamWithoutTranscodingEx(url, connection) {
    const apiType = this.getApiTypeFromStartRtmpStreamWithoutTranscodingEx(url, connection);
    const jsonParams = {
      url: url,
      connection: connection,
      toJSON: () => {
        return {
          url: url,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRtmpStreamWithoutTranscodingEx(url, connection) {
    return 'RtcEngineEx_startRtmpStreamWithoutTranscodingEx';
  }
  startRtmpStreamWithTranscodingEx(url, transcoding, connection) {
    const apiType = this.getApiTypeFromStartRtmpStreamWithTranscodingEx(url, transcoding, connection);
    const jsonParams = {
      url: url,
      transcoding: transcoding,
      connection: connection,
      toJSON: () => {
        return {
          url: url,
          transcoding: transcoding,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartRtmpStreamWithTranscodingEx(url, transcoding, connection) {
    return 'RtcEngineEx_startRtmpStreamWithTranscodingEx';
  }
  updateRtmpTranscodingEx(transcoding, connection) {
    const apiType = this.getApiTypeFromUpdateRtmpTranscodingEx(transcoding, connection);
    const jsonParams = {
      transcoding: transcoding,
      connection: connection,
      toJSON: () => {
        return {
          transcoding: transcoding,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateRtmpTranscodingEx(transcoding, connection) {
    return 'RtcEngineEx_updateRtmpTranscodingEx';
  }
  stopRtmpStreamEx(url, connection) {
    const apiType = this.getApiTypeFromStopRtmpStreamEx(url, connection);
    const jsonParams = {
      url: url,
      connection: connection,
      toJSON: () => {
        return {
          url: url,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopRtmpStreamEx(url, connection) {
    return 'RtcEngineEx_stopRtmpStreamEx';
  }
  startOrUpdateChannelMediaRelayEx(configuration, connection) {
    const apiType = this.getApiTypeFromStartOrUpdateChannelMediaRelayEx(configuration, connection);
    const jsonParams = {
      configuration: configuration,
      connection: connection,
      toJSON: () => {
        return {
          configuration: configuration,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartOrUpdateChannelMediaRelayEx(configuration, connection) {
    return 'RtcEngineEx_startOrUpdateChannelMediaRelayEx';
  }
  startChannelMediaRelayEx(configuration, connection) {
    const apiType = this.getApiTypeFromStartChannelMediaRelayEx(configuration, connection);
    const jsonParams = {
      configuration: configuration,
      connection: connection,
      toJSON: () => {
        return {
          configuration: configuration,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartChannelMediaRelayEx(configuration, connection) {
    return 'RtcEngineEx_startChannelMediaRelayEx';
  }
  updateChannelMediaRelayEx(configuration, connection) {
    const apiType = this.getApiTypeFromUpdateChannelMediaRelayEx(configuration, connection);
    const jsonParams = {
      configuration: configuration,
      connection: connection,
      toJSON: () => {
        return {
          configuration: configuration,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromUpdateChannelMediaRelayEx(configuration, connection) {
    return 'RtcEngineEx_updateChannelMediaRelayEx';
  }
  stopChannelMediaRelayEx(connection) {
    const apiType = this.getApiTypeFromStopChannelMediaRelayEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStopChannelMediaRelayEx(connection) {
    return 'RtcEngineEx_stopChannelMediaRelayEx';
  }
  pauseAllChannelMediaRelayEx(connection) {
    const apiType = this.getApiTypeFromPauseAllChannelMediaRelayEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromPauseAllChannelMediaRelayEx(connection) {
    return 'RtcEngineEx_pauseAllChannelMediaRelayEx';
  }
  resumeAllChannelMediaRelayEx(connection) {
    const apiType = this.getApiTypeFromResumeAllChannelMediaRelayEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromResumeAllChannelMediaRelayEx(connection) {
    return 'RtcEngineEx_resumeAllChannelMediaRelayEx';
  }
  getUserInfoByUserAccountEx(userAccount, connection) {
    const apiType = this.getApiTypeFromGetUserInfoByUserAccountEx(userAccount, connection);
    const jsonParams = {
      userAccount: userAccount,
      connection: connection,
      toJSON: () => {
        return {
          userAccount: userAccount,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const userInfo = jsonResults.userInfo;
    return userInfo;
  }
  getApiTypeFromGetUserInfoByUserAccountEx(userAccount, connection) {
    return 'RtcEngineEx_getUserInfoByUserAccountEx';
  }
  getUserInfoByUidEx(uid, connection) {
    const apiType = this.getApiTypeFromGetUserInfoByUidEx(uid, connection);
    const jsonParams = {
      uid: uid,
      connection: connection,
      toJSON: () => {
        return {
          uid: uid,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    const userInfo = jsonResults.userInfo;
    return userInfo;
  }
  getApiTypeFromGetUserInfoByUidEx(uid, connection) {
    return 'RtcEngineEx_getUserInfoByUidEx';
  }
  enableDualStreamModeEx(enabled, streamConfig, connection) {
    const apiType = this.getApiTypeFromEnableDualStreamModeEx(enabled, streamConfig, connection);
    const jsonParams = {
      enabled: enabled,
      streamConfig: streamConfig,
      connection: connection,
      toJSON: () => {
        return {
          enabled: enabled,
          streamConfig: streamConfig,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableDualStreamModeEx(enabled, streamConfig, connection) {
    return 'RtcEngineEx_enableDualStreamModeEx';
  }
  setDualStreamModeEx(mode, streamConfig, connection) {
    const apiType = this.getApiTypeFromSetDualStreamModeEx(mode, streamConfig, connection);
    const jsonParams = {
      mode: mode,
      streamConfig: streamConfig,
      connection: connection,
      toJSON: () => {
        return {
          mode: mode,
          streamConfig: streamConfig,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetDualStreamModeEx(mode, streamConfig, connection) {
    return 'RtcEngineEx_setDualStreamModeEx';
  }
  setHighPriorityUserListEx(uidList, uidNum, option, connection) {
    const apiType = this.getApiTypeFromSetHighPriorityUserListEx(uidList, uidNum, option, connection);
    const jsonParams = {
      uidList: uidList,
      uidNum: uidNum,
      option: option,
      connection: connection,
      toJSON: () => {
        return {
          uidList: uidList,
          uidNum: uidNum,
          option: option,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromSetHighPriorityUserListEx(uidList, uidNum, option, connection) {
    return 'RtcEngineEx_setHighPriorityUserListEx';
  }
  takeSnapshotEx(connection, uid, filePath) {
    const apiType = this.getApiTypeFromTakeSnapshotEx(connection, uid, filePath);
    const jsonParams = {
      connection: connection,
      uid: uid,
      filePath: filePath,
      toJSON: () => {
        return {
          connection: connection,
          uid: uid,
          filePath: filePath
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromTakeSnapshotEx(connection, uid, filePath) {
    return 'RtcEngineEx_takeSnapshotEx';
  }
  enableContentInspectEx(enabled, config, connection) {
    const apiType = this.getApiTypeFromEnableContentInspectEx(enabled, config, connection);
    const jsonParams = {
      enabled: enabled,
      config: config,
      connection: connection,
      toJSON: () => {
        return {
          enabled: enabled,
          config: config,
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromEnableContentInspectEx(enabled, config, connection) {
    return 'RtcEngineEx_enableContentInspectEx';
  }
  startMediaRenderingTracingEx(connection) {
    const apiType = this.getApiTypeFromStartMediaRenderingTracingEx(connection);
    const jsonParams = {
      connection: connection,
      toJSON: () => {
        return {
          connection: connection
        };
      }
    };
    const jsonResults = callIrisApi.call(this, apiType, jsonParams);
    return jsonResults.result;
  }
  getApiTypeFromStartMediaRenderingTracingEx(connection) {
    return 'RtcEngineEx_startMediaRenderingTracingEx';
  }
}
import { callIrisApi } from '../internal/IrisApiEngine';
