// react-native-agora/lib/commonjs/IAgoraSpatialAudio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SpatialAudioZone = exports.RemoteVoicePositionInfo = exports.ILocalSpatialAudioEngine = exports.IBaseSpatialAudioEngine = void 0;
require("./extension/IAgoraSpatialAudioExtension");
/**
 * The spatial position of the remote user or the media player.
 */
class RemoteVoicePositionInfo {}

/**
 * Sound insulation area settings.
 */
exports.RemoteVoicePositionInfo = RemoteVoicePositionInfo;
class SpatialAudioZone {}

/**
 * This class contains some of the APIs in the ILocalSpatialAudioEngine class.
 *
 * The ILocalSpatialAudioEngine class inherits from IBaseSpatialAudioEngine.
 */
exports.SpatialAudioZone = SpatialAudioZone;
class IBaseSpatialAudioEngine {}

/**
 * This class calculates user positions through the SDK to implement the spatial audio effect.
 *
 * This class inherits from IBaseSpatialAudioEngine. Before calling other APIs in this class, you need to call the initialize method to initialize this class.
 */
exports.IBaseSpatialAudioEngine = IBaseSpatialAudioEngine;
class ILocalSpatialAudioEngine extends IBaseSpatialAudioEngine {}
exports.ILocalSpatialAudioEngine = ILocalSpatialAudioEngine;
